synapse_data_warehouse/synapse/dynamic_tables/V2.22.0__accessrequirement_latest.sql

```diff
@@ -0,0 +1,22 @@
+use schema {{database_name}}.synapse; --noqa: JJ01,PRS,TMP,CP01
+
+CREATE DYNAMIC TABLE IF NOT EXISTS ACCESSREQUIREMENT_LATEST
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
+    AS
+    WITH latest_unique_rows AS (
+        SELECT
+            accessrequirementsnapshots.*,
+        FROM
+            {{database_name}}.synapse_raw.accessrequirementsnapshots --noqa: TMP
+        QUALIFY ROW_NUMBER() OVER (
```

Comment:
This is nice! Please do add in the snapshot_timestamp < 2 weeks check

Reply:
I did not add the timestamp restriction because `accessrequirementsnapshots` records deletions:

<img width="622" alt="image" src="https://github.com/user-attachments/assets/6b98d154-8da8-4071-88dd-244f5cd98e83">

For my understanding, we add these timestamp windows because:
* It reduces the scope of the query, making the CTE more efficient
* For tables that do not capture the deletion of a record, it reduces the chance that a record shows up in `latest` even though it's deleted from the system. For example, if a record was deleted but `change_type = 'DELETE'` is not supported in the table, setting the 2-week window minimizes the chance of it showing up in the final table, assuming the record wasn't deleted within those 2 weeks.

Let me know if this is correct

Reply:
Ooo thanks for checking! This then becomes a bit more complicated! 

How do we use those DELETE events to make sure we are removing the ones that should be removed?

Reply:
Ah I think 7 days may be too short because there might be records that cut in and out of the snapshot dates, that's why I chose 14 or in some cases 30.

I think that should work if you make sure N=1 isn't a DELETE event but we should test that theory.

I do like your thought of tracking delete events but thats not what's in the latest tables currently.

Reply:
The delete events could also be in their own tables - I think that would make more sense.

So just to clarify once more:
* For snapshots that record DELETE events, we _don't_ need a timestamp window because we can filter out deleted items with `change_type`
* For snapshots that do not record DELETE events, we use a timestamp window to filter out deleted items, so long as the deletion happened outside of the timestamp window
* The other advantage is using a timestamp window improves query efficiency since you're pulling in less data in your CTE

So for this case, since I can filter out the deleted records using `change_type`, I don't _need_ to use a timestamp window, but I can use one for query efficiency.

Reply:
Great summary! Thanks!

Reply:
perfect, thanks!

cc @philerooski @danlu1 so they know too

Reply:
Updated https://github.com/Sage-Bionetworks/snowflake/pull/84/commits/3cbc2d6e1c9f2a921644b60369664b822565bbef